File: snippets/fsharp/System/String/Substring/Substring2.fs
module Substring2.fs
// <Snippet2>
let s = "aaaaabbbcccccccdd"
let charRange = 'b'
let startIndex = s.IndexOf charRange
let endIndex = s.LastIndexOf charRange
let length = endIndex - startIndex + 1
printfn $"{s}.Substring({startIndex}, {length}) = {s.Substring(startIndex, length)}"

// The example displays the following output:
//       aaaaabbbcccccccdd.Substring(5, 3) = bbb
// </Snippet2>

File: snippets/csharp/System/String/Substring/Substring2.cs
using System;

public class SubstringRangeExample
{
    public static void Run()
    {
        // <Snippet2>
        string s = "aaaaabbbcccccccdd";
        char charRange = 'b';
        int startIndex = s.IndexOf(charRange);
        int endIndex = s.LastIndexOf(charRange);
        int length = endIndex - startIndex + 1;
        Console.WriteLine($"{s}.Substring({startIndex}, {length}) = {s.Substring(startIndex, length)}");

        // The example displays the following output:
        //       aaaaabbbcccccccdd.Substring(5, 3) = bbb
        // </Snippet2>
    }
}
